Dependencies/Onboarding/Sources/Onboarding/Views/Launch/Login/LoginViewModel.swift

```diff
@@ -5,24 +5,81 @@
 //  Created by MaTooSens on 19/08/2023.
 //
 
+import AuthenticationInterface
 import CoreInterface
-import Foundation
+import DependencyInjection
+import SwiftUI
 
-final class LoginViewModel: ViewModelInterface {
-    enum State {
-        case idle
-        case loading
-        case error
-        case loaded
+final class LoginViewModel: ObservableObject, AsyncInterface {
+    enum Event {
+        case nextField
     }
     
-    enum Event {
-       // event
+    enum AsyncEvent{
+        case signIn
     }
     
-    @Published private(set) var state: State = .idle
+    @Inject private var authenticationManager: AuthenticationManagerInterface
+    
+    @Published var focusedField: LoginView.Field?
+    @Published var email: String = ""
+    @Published var password: String = ""
     
     func trigger(_ event: Event) {
-        // action 
+        switch event {
+        case .nextField:
+            nextField()
+        }
+    }
+    
+    func trigger(_ event: AsyncEvent) async throws {
+        switch event {
+        case .signIn:
+            try await signIn()
+        }
+    }
+}
+
+
+// MARK: Functions
+extension LoginViewModel {
+    private func signIn() async throws {
+        _ = try await authenticationManager.signIn(email: email, password: password)
+    }
+}
+
+
+// MARK: Focus TextFields Functions
+extension LoginViewModel {
+    func focus(_ field: LoginView.Field) -> Binding<Bool> {
+        Binding.constant(focusedField == field)
+    }
+    
+    private func nextField() {
+        var fields: [LoginView.Field] = []
+        
+        if email.isEmpty || focusedField == .email { fields.append(.email) }
+        if password.isEmpty || focusedField == .password { fields.append(.password) }
+        
+        guard
+            let currentField = focusedField,
+            let index = fields.firstIndex(of: currentField)
+        else {
+            focusedField = nil
+            return
+        }
+        
+        let lastIndex = fields.endIndex - 1
+        
+        let areNotEmpty = !email.isEmpty && !password.isEmpty
+        
+        if lastIndex > 0 {
+            let nextFieldIndex = index < lastIndex ? index + 1 : 0
+            focusedField = fields[nextFieldIndex]
```

Comment:
tu bym nie tworzył dodatkowej zmiennej chyba
```suggestion
            focusedField = fields[index < lastIndex ? index + 1 : 0]
```